torch_geometric/utils/trim_to_layer.py

```diff
@@ -0,0 +1,104 @@
+from typing import Dict, Tuple, Union
+
+import torch
+from torch import Tensor
+from torch_sparse import SparseStorage
+
+from torch_geometric.typing import Adj, EdgeType, NodeType, SparseTensor
+
+
+def trim_to_i_layer(
```

Comment:
I think this is the function we would want to keep in the first iteration.

I think the interface should be something like
```python
def trim_to_layer(
    layer: int,
    edge_index: Tensor,
    node_attrs: Union[Tensor, List[Tensor]],
    edge_attrs: Union[Tensor, List[Tensor]],
    num_nodes_per_layer: List[int],
    num_edges_per_layer: List[int],
):
```
such that we can support both node and edge features, and possibly multiple of them (e.g., narrowing down both `x` and `pos`).

Otherwise, this LGTM. If you want to desperately include `SparseTensor` code here, I think it is the most straightforward way to make use of the `torch_sparse.narrow` functionality to implement this. Let me know if you see any blockers with this.

Reply:
@rusty1s updated PR accordingly to discussed changes.